lib/di/state_with_bloc.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:where_to_have_lunch/di/injector.dart';
+import 'package:where_to_have_lunch/utils/logger.dart';
+
+abstract class StateWithBloC<W extends StatefulWidget, STATE extends Equatable, B extends BlocBase<STATE>> extends State<W> {
```

Comment:
Base State Class that helps with the dependency injection and wraps also the `BlocProvider` and `BlocConsumer`